src/optics/lib/Convex/CardanoApi/Lenses.hs

```diff
@@ -208,6 +218,24 @@ instance IsBabbageEraOnwards C.BabbageEra where
 instance IsBabbageEraOnwards C.ConwayEra where
    babbageEraOnwards = C.BabbageEraOnwardsConway
 
+txBody :: Getter (C.Tx era) (C.TxBody era)
+txBody = L.to get_ where
+  get_ (C.Tx txb _) = txb
+
+txBodyContent :: Getter (C.TxBody era) (C.TxBodyContent C.ViewTx era)
+txBodyContent = L.to get_ where
+  get_ (C.TxBody txbc) = txbc
+
+txBodyScriptData :: Getter (C.TxBody era) (C.TxBodyScriptData era)
```

Comment:
Same here and for the remaining setters

Reply:
This is a bit problematic for implementing a setter. It's doable for setting the datum. However, setting the redeemer does not make sense, because you need to provide the `Redeemers` type which requires the `Redeemer` to be associated to some `ExUnit`. However, to get the `ExUnit`, we need to get it from the script witnesses. I don't think it makes sense to add the script witness as a parameter.